### ISSUE TITLE
Problems finding dune-common when installing opm-autodiff

### ISSUE THREAD

**Issue by TobiasRitschel** [OPM/opm-autodiff#353]
Hi

I am trying to install opm-autodiff from source. I have installed opm-parser and opm-core from source and installed dune-cornerpoint with sudo apt-get install dune-cornerpoint.

When I try to cmake from my sibling build directory it fails to configure. From the output to the terminal I noticed the following lines
```
[...]
-- Performing Test HAVE_REGEX
-- Performing Test HAVE_REGEX - Failed
[...]
-- Finding package dune-common using module mode
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.26") 
-- Failed to find library "dunecommon" for module dune-common
[...]
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
[...]
-- Performing Test HAVE_DUNE_COMMON
-- Performing Test HAVE_DUNE_COMMON - Success
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find dune-common (missing: dune-common_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/OpmPackage.cmake:358 (find_package_handle_standard_args)
  cmake/Modules/Finddune-common.cmake:14 (find_opm_package)
  cmake/Modules/OpmFind.cmake:147 (find_package)
  cmake/Modules/OpmFind.cmake:211 (find_and_append_package_to)
  cmake/Modules/OpmLibMain.cmake:104 (find_and_append_package_list_to)
  CMakeLists.txt:74 (include)


-- Configuring incomplete, errors occurred!
See also "/home/tobias/Documents/OPM/opm-autodiff-build/CMakeFiles/CMakeOutput.log".
See also "/home/tobias/Documents/OPM/opm-autodiff-build/CMakeFiles/CMakeError.log".
```
It seems to me that it is having problems finding the dune-common library files. `CMakeOutput.log` does not seem to contain any errors whereas `CMakeError.log` mentions the `pthreads` not found. Calling `locate libdune-common` gives
```
/usr/share/doc/libdune-common-2.2.1
/usr/share/doc/libdune-common-dev
/usr/share/doc/libdune-common-2.2.1/changelog.Debian.gz
/usr/share/doc/libdune-common-2.2.1/copyright
/usr/share/doc/libdune-common-dev/changelog.Debian.gz
/usr/share/doc/libdune-common-dev/copyright
/usr/share/lintian/overrides/libdune-common-2.2.1
/var/lib/dpkg/info/libdune-common-2.2.1:amd64.list
/var/lib/dpkg/info/libdune-common-2.2.1:amd64.md5sums
/var/lib/dpkg/info/libdune-common-2.2.1:amd64.postinst
/var/lib/dpkg/info/libdune-common-2.2.1:amd64.postrm
/var/lib/dpkg/info/libdune-common-2.2.1:amd64.shlibs
/var/lib/dpkg/info/libdune-common-dev.list
/var/lib/dpkg/info/libdune-common-dev.md5sums
```
I would think that there should be some library files, however, I've installed this (and others) using
```
sudo apt-get install libdune-common-dev libdune-geometry-dev \
  libdune-istl-dev libdune-grid-dev
```
I also tried to install `dune-cornerpoint` from source with the same problem and thought that using `apt-get` was easier.

Any help will be much appreciated

Kind regards
Tobias

**Comment by atgeirr** [comment]
You seem to have a problem with dune-common, after installing it with apt-get. This *should* work, but I am not sure what goes wrong here. However I did notice one thing:

> I am trying to install opm-autodiff from source. I have installed opm-parser and opm-core from source and installed dune-cornerpoint with sudo apt-get install dune-cornerpoint.

That will not work. Despite the naming, dune-cornerpoint is an opm module (providing a class CpGrid that implements a corner-point grid with the Dune grid interface), and you must install that module in the same way as the other opm modules (i.e. from source).

Did you use (old) release tarballs or clone with git?

**Comment by TobiasRitschel** [comment]
Okay I will install it from source. I use git. For as much as possible, I try to follow the instructions given on the Download page of the website (http://www.opm-project.org/download.php). Is there supposed to be dune-common library files located somewhere? I get the same problem trying to install dune-cornerpoint.

**Comment by bska** [comment]
> locate libdune-common gives
[... list of files omitted ...]
I would think that there should be some library files

Yes, you would think so.  However, the actual library files are called `libdune<module>` not `libdune-<module>` (note missing hyphen).  You need to run the command
```sh
locate libdunecommon
```
to find the location of the dune-common library files.

**Comment by TobiasRitschel** [comment]
I see. In this case it didn't return anything. I seem to remember that I might have manually deleted the library files when I was trying to solve another problem (https://github.com/OPM/opm-core/issues/780#issuecomment-92033759). I tried removing all the libdune and opm modules installed and reinstalling.

Now it cannot find libdune-grid giving similar error messages as above. "locate libdunecommon" as well as "locate libdunegrid" and others do not return anything.

Is there an alternative way to get the library files I need?

**Comment by akva2** [comment]
locate uses a cache that is by default updated once per day. try find / 
-name libdunegrid*

in any case the libraries will be located in /usr/lib/<arch>/

**Comment by TobiasRitschel** [comment]
Ah, that's right. "find / -name libdunegrid*" confirms that libdunegeometry.so (it wasn't libdunegrid.so as I said). All relevant shared libraries seem to be in the folder /usr/lib/x86_64-linux-gnu. When I type "ccmake ../dune-cornerpoint" I can see that dune-geometry_LIBRARY is set to /usr/lib/x86_64-linux-gnu/libdunegeometry.so and dune-grid_LIBRARY to /usr/lib/x86_64-linux-gnu/libdunegrid.so.

When I press "c" to configure it gives the error message

 CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
   Could NOT find dune-geometry (missing: HAVE_DUNE_GEOMETRY)
 Call Stack (most recent call first):
   /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
   cmake/Modules/OpmPackage.cmake:358 (find_package_handle_standard_args)
   cmake/Modules/Finddune-geometry.cmake:14 (find_opm_package)
   cmake/Modules/OpmFind.cmake:147 (find_package)
   cmake/Modules/OpmPackage.cmake:269 (find_and_append_package_to)
   cmake/Modules/Finddune-grid.cmake:17 (find_opm_package)
   cmake/Modules/OpmFind.cmake:147 (find_package)
   cmake/Modules/OpmFind.cmake:211 (find_and_append_package_to)
   cmake/Modules/OpmLibMain.cmake:104 (find_and_append_package_list_to)
   CMakeLists.txt:77 (include)

I also tried adding /usr/lib/x86_64-linux-gnu to LD_LIBRARY_PATH and it is already a part of LIBRARY_PATH but I guess that's not relevant in this case.

**Comment by akva2** [comment]
did you install the -dev packages ?

**Comment by TobiasRitschel** [comment]
Yes, I believe so.

**Comment by TobiasRitschel** [comment]
I now deleted all dune-cornerpoint files (the directory cloned from git and the corresponding build directory. I also had no opm-autodiff directories). I installed dune-cornerpoint and opm-autodiff using the instructions from the Building from source part of the Download page of http://www.opm-project.org/download.php and I am now able to compile and run sim_2p_incomp_ad.cpp. I guess something needed to be reset or deleted.

Thanks for the help.
